var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -20,13 +20,24 @@ class Zfp(CMakePackage):
     version('0.5.2', sha256='89e718edb966422b5898b5c37b1b0a781d4effacb511520558469e3ff7f65d7c')
     version('0.5.1', sha256='867c04cf965f1c70d9725b396c6e1b5d29db55b0d69b8e87a995aaebd221b830')
 
+
+
     variant('bsws',
             default='64',
             values=('8', '16', '32', '64'),
             multi=False,
             description='Bit stream word size: use smaller for finer '
             'rate granularity. Use 8 for H5Z-ZFP filter.')
-
+    variant('strided', default=False,
```

Comment:
Are you aware of use-cases where any/all of these features are actually needed? Reason I ask is that I think speculative variant addition is not the best practice in Spack-world because it makes for unnecessarily long/complicated specs. OTOH, I haven't been tracking recent Spack activities enough to be an authority on this topic. Its only a suspicion based on past experiences.

Reply:
I think it depends on your own personal philosophy. I tend to add variants for everything because I think it's better than some random person adding a new variant every month. At least if I add them all at once there are fewer rebuilds.

Reply:
I thought: if it is in the public CMake options, it will be exposed for a reason. If we as packagers do not expose a feature it will not be usable downstream. These options also look stable to me in all tagged version.
I keep the defaults as in CMake, so one does not see them in "daily CLI work" that much.

Reply:
OTOH, writing well-controlled spack packages triggers the developer in me:
"Is it really useful to expose that many options as individual compile-time variants of my code? Should I be a bit more careful with defines and build variations that I could also expose in another way, which leads to less fragmentation of my binary flavors?" - I mean, most other package managers will have to decide for one of these variants and a large part of my functionality might never see the light of day that way...